aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -46,6 +46,7 @@ Tensor& eye_out_cuda(Tensor& result, int64_t n, int64_t m) {
 Tensor empty_cuda(IntArrayRef size, const TensorOptions& options) {
   AT_ASSERT(options.backend() == at::Backend::CUDA);
   AT_ASSERT(!options.is_variable());  // is_variable should have been 'unpacked'  // TODO: remove this when Variable and Tensor are merged
+  AT_CHECK(!options.pinned_memory(), "Only dense CPU tensors can be pinned");
```

Comment:
this error message is a little strange.  Does density have anything to do with it?  Presumably we could have sparse layout pinned tensors.  Also I would try to avoid using the word 'dense' (we aren't consistent about this), but the correct term is really `strided` because that's the name of the layout.

Reply:
The message written this way because of the https://github.com/pytorch/pytorch/blob/d74b11ce0ed33a13bb6befff03c83a3d115255b1/aten/src/ATen/native/Memory.cpp#L12

If you blame it, you will see that the previous message was confusing in case of attempt of pinning sparse CPU tensor. 

I'm totally fine to change error test to the: `Only strided CPU tensors can be pinned` or anything similar (but proper way would be implementing pin_memory() call on sparse tensors ;) )